File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseUserCardModel.cs
using KnowledgeSharingApi.Domains.Enums;
using KnowledgeSharingApi.Domains.Models.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseUserCardModel : Entity
    {
        public Guid UserId { get; set; }

        public string Username { get; set; } = string.Empty;

        public string? Avatar { get; set; }

        public string? Cover { get; set; }

        public string FullName { get; set; } = string.Empty;

        public string? Email { get; set; }

        public string? PhoneNumber { get; set; }

        public string Role { get; set; } = string.Empty;

        public string? Bio { get; set; }

        public EUserRelationType? UserRelationType { get; set; }

        public Guid? UserRelationId { get; set; }

        public ECourseRoleType? CourseRoleType { get; set; }

        public Guid? CourseRelationId { get; set; }

        protected override ResponseUserCardModel Init()
        {
            return new ResponseUserCardModel();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-user-card-model.js

import Entity from '../entity.js';

class ResponseUserCardModel extends Entity {
    constructor() {
        super();
        this.UserId = null;
        this.Username = null;
        this.Email = null;
        this.PhoneNumber = null;
        this.Avatar = null;
        this.Cover = null;
        this.FullName = null;
        this.Role = null;
        this.Bio = null;
        this.UserRelationType = null;
        this.UserRelationId = null;
        this.CourseRoleType = null;
        this.CourseRelationId = null;
    }

    init() {
        return new ResponseUserCardModel();
    }
}

export default ResponseUserCardModel;
